Documentation/api-reference.md

```diff
@@ -0,0 +1,122 @@
+# Adobe Experience Platform Consent for Edge Network Extension - Android
+
+## Prerequisites
+
+Refer to the [Getting Started Guide](getting-started.md)
+
+## API reference
+
+- [extensionVersion](#extensionversion)
+- [getConsents](#getConsents)
+- [registerExtension](#registerextension)
+- [updateConsents](#updateConsents)
+------
+
+### extensionVersion
+
+The extensionVersion() API returns the version of the client-side Consent extension.
+
+#### Java
+
+##### Syntax
+```java
+public static String extensionVersion()
+```
+
+##### Example
+```java
+String extensionVersion = Consent.extensionVersion();
+```
+
+------
+
+### getConsents
+
+Retrieves the current consent preferences stored in the Consent extension.
+
+#### Java
+
+##### Syntax
+```java
+public static void getConsents(final AdobeCallback<Map<String, Object>> callback);
+```
+* callback - callback invoked with the current consents of the extension. If an AdobeCallbackWithError is provided, an AdobeError, can be returned in the eventuality of any error that occurred while getting the user consents. The callback may be invoked on a different thread.
+
+##### Example
+```java
+Consent.getConsents(new AdobeCallback<Map<String, Object>>() {
+    @Override
+    public void call(Map<String, Object> currentConsents) {
+        // handle currentConsents
+    }
+});
+```
+
+------
+
+### registerExtension
+
+Registers the Edge Consent extension with the Mobile Core SDK.
+
+> **Warning**
+> Edge Consent extension registerExtension API is deprecated. Refer to [Mobile Core API documentation](https://github.com/adobe/aepsdk-core-android/blob/staging/docs/Usage/MobileCore.md#registering-extensions-and-starting-the-sdk) for registering extension.
```

Comment:
Maybe more simply, "Deprecated as of 2.0.0. See MobileCore.registerExtensions API", with the link around "MobileCore.registerExtensions"?